src/SoftRobots/component/constraint/model/CableModel.inl

```diff
@@ -212,15 +256,34 @@ void CableModel<DataTypes>::internalInit()
 {
     initActuatedPoints();
     checkIndicesRegardingState();
+
+    const unsigned int id_method = d_method.getValue().getSelectedId();
+    if (id_method == 1 || id_method == 2)
+        initCableActionAreas();
 }
 
 
 template<class DataTypes>
 void CableModel<DataTypes>::initActuatedPoints()
-{
-    int nbIndices = d_indices.getValue().size();
+{   
+    auto indices = sofa::helper::getWriteOnlyAccessor(d_indices);
+    int nbIndices = indices.size();
     ReadAccessor<Data<VecCoord>> positions = m_state->readPositions();
-    const SetIndexArray &indices = d_indices.getValue();
+    ReadAccessor<Data<VecCoord>> centers = d_centers;
+    unsigned int nbCenters = centers.size();
+    m_hasCenters = false;
```

Comment:
Why having this variable if you have the size of `d_centers`? It is dangerous if `d_centers` changes during the simulation

Reply:
Same comment again